yarpctest/recorder/recorder.go

```diff
@@ -0,0 +1,317 @@
+package recorder
+
+import (
+	"bytes"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"hash/fnv"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"unicode"
+
+	"github.com/yarpc/yarpc-go/transport"
+	"golang.org/x/net/context"
+	"gopkg.in/yaml.v2"
+)
+
+var recorderFlag = flag.String("recorder", "replay",
+	`replay: replay from recorded request/response pairs.
+overwrite: record all request/response pairs, overwriting records.
+append: replay existing and record new request/response pairs.`)
+
+// Recorder records & replay yarpc requests on the client side.
+//
+// For recording, the client must be connected and able to issue requests to a
+// remote service. Every request and its response is recorded into a YAML file,
+// under the directory "testdata/recordings".
+//
+// During replay, the client doesn't need to be connected, for any recorded
+// request Recorder will return the recorded response. Any new request will
+// return an error.
+type Recorder struct {
```

Comment:
Yes but! It will then add the --recorder flag even if you do not use the recorder itself...